src/components/vega-lite/index.tsx

```diff
@@ -3,27 +3,44 @@ import * as vega from 'vega';
 import * as vl from 'vega-lite';
 import {TopLevelExtendedSpec} from 'vega-lite/build/src/spec';
 import * as vegaTooltip from 'vega-tooltip';
+import {Logger} from '../../models/shelf/logger';
+import {Warnings} from './warnings';
 
 export interface VegaLiteProps {
   spec: TopLevelExtendedSpec;
 
   renderer?: 'svg' | 'canvas';
+
+  logger: Logger;
```

Comment:
Currently logger is a singleton logger stored in the global redux store, but it is render in all VegaLite plots instance.  This means that if we render a gallery of plots all of them will display redundant warning messages.

We should discuss a design we want but in any case, we should pick a model that is consistent with the displays.

a) Local logger for each plot and displays the plot warning independent for each plot
or b) Global logger and displays the warning as a separate component.